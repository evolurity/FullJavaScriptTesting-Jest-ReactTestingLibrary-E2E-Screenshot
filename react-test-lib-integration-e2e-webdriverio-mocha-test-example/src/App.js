// import './App.css';
// import {useEffect, useState} from "react";
//
// function App() {
//     const [data, setData] = useState(null);
//     const [toggle, setToggle] = useState(false);
//
//     const [value, setValue] = useState('')
//
//     const onClick = () => {
//         setToggle(prevState => !prevState);
//     }
//
//     useEffect(() => {
//         setTimeout(() => {
//             setData({})
//         }, 100)
//     }, [])
//
//
//     return (
//         <div className="App">
//             <h1 data-testid={'value-elem'}>{value}</h1>
//             {toggle && <div data-testid={"toggle-elem"}>toggle</div>}
//             {data && <div style={{color: "red"}}>data</div>}
//             <h1>Simple App for testing</h1>
//             <button data-testid={"toggle-btn"} onClick={onClick}>Click me</button>
//             <input onInput={(event => setValue(event.target.value))} data-testid={"input-field"} type="text" placeholder={"test text"}/>
//         </div>
//     );
// }
//
// export default App;

import React, {Component} from 'react';
import AppRouter from "./routes/AppRouter";
import Navbar from "./components/Navbar/Navbar";

class App extends Component {
    render() {
        return (
            <div>
                <Navbar/>
                <AppRouter/>
            </div>
        );
    }
}

export default App;