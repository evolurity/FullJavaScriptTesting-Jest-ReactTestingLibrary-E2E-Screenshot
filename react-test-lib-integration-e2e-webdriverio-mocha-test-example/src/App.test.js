import {fireEvent, render, screen} from '@testing-library/react';
import App from './App';
import userEvent from "@testing-library/user-event";


describe('TEST APP', () => {

    test('example 1', () => {
        render(<App/>);
        const titleElement = screen.getByText(/simple app for testing/i);
        const btn = screen.getByRole('button');
        const input = screen.getByPlaceholderText(/test text/i);
        expect(titleElement).toBeInTheDocument();
        expect(btn).toBeInTheDocument();
        expect(input).toBeInTheDocument();
        expect(input).toMatchSnapshot();
    });


    test('example 2', async () => {
        render(<App/>);
        const titleElement = screen.queryByText(/not found element/i);
        expect(titleElement).toBeNull();
        const data = await screen.findByText(/data/i);
        expect(data).toBeInTheDocument();
        expect(data).toHaveStyle({color: 'red'});
        expect(data).toMatchSnapshot();
    });

    test('example 3: CLICK EVENT', () => {
        render(<App/>);
        const btn = screen.getByTestId('toggle-btn');
        expect(screen.queryByTestId('toggle-elem')).toBeNull();
        fireEvent.click(btn);
        expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
        fireEvent.click(btn);
        expect(screen.queryByTestId('toggle-elem')).toBeNull();
    })

    test('example 4: INPUT EVENT', () => {
        render(<App/>);
        const input = screen.getByPlaceholderText(/test text/i);
        expect(screen.queryByTestId('value-elem')).toContainHTML('');
        // Искусственное событие
        // fireEvent.input(input, {
        //   target: { value: '123test'}
        // })
        // Полноценная эмуляция действия пользователя
        userEvent.type(input, '123test')
        expect(screen.queryByTestId('value-elem')).toContainHTML('123test');
    })
})
