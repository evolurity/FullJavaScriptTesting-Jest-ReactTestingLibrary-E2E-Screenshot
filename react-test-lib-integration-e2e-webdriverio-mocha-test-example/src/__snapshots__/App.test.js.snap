// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP renders learn react link 1`] = `
<input
  placeholder="test text"
  type="text"
/>
`;

exports[`TEST APP renders learn react link 2 1`] = `
<div
  style="color: red;"
>
  data
</div>
`;
